### ISSUE TITLE
[BUG] plugin prevents dart analyze running without an internet connection

### ISSUE THREAD

**Issue by saty9** [dart-code-checker/dart-code-metrics#976]
* **Dart code metrics version:**
4.16.0
* **Dart sdk version:**
2.17.6

![image](https://user-images.githubusercontent.com/19638136/186850711-a2c01ff7-9c34-4e07-b159-17b65be000fb.png)

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
# This file configures the analyzer, which statically analyzes Dart code to
# check for errors, warnings, and lints.
#
# The issues identified by the analyzer are surfaced in the UI of Dart-enabled
# IDEs (https://dart.dev/tools#ides-and-editors). The analyzer can also be
# invoked from the command line by running `flutter analyze`.

# The following line activates a set of recommended lints for Flutter apps,
# packages, and plugins designed to encourage good coding practices.
include: package:flutter_lints/flutter.yaml

linter:
  # The lint rules applied to this project can be customized in the
  # section below to disable rules from the `package:flutter_lints/flutter.yaml`
  # included above or to enable additional rules. A list of all available lints
  # and their documentation is published at
  # https://dart-lang.github.io/linter/lints/index.html.
  #
  # Instead of disabling a lint rule for the entire project in the
  # section below, it can also be suppressed for a single line of code
  # or a specific dart file by using the `// ignore: name_of_lint` and
  # `// ignore_for_file: name_of_lint` syntax on the line or in the file
  # producing the lint.
  rules:
    # avoid_print: false  # Uncomment to disable the `avoid_print` rule
    # prefer_single_quotes: true  # Uncomment to enable the `prefer_single_quotes` rule
    require_trailing_commas: true

# Additional information about this file can be found at
# https://dart.dev/guides/language/analysis-options


analyzer:
  plugins:
    - dart_code_metrics

dart_code_metrics:
  rules:
    avoid-use-expanded-as-spacer: true
    always-remove-listener: true
    avoid-border-all: true
```

</details>

**What did you do? Please include the source code example causing the issue.**
 Turned off my devices internet connect / went somewhere with no internet connection

**What did you expect to happen?**

`dart analyze` to run and produce a list of issues

**What actually happened?**

`dart analyze` hangs and doesnt terminate

I then commented out the plugin and its config and dart analyze works offline again

**Are you willing to submit a pull request to fix this bug?**

No

**Comment by incendial** [comment]
I'm not sure that this issue belongs to DCM, since this api is out of our scope. Any chance you can provide a repro?

**Comment by saty9** [comment]
what would a reproduction look like? do you just want a template project with this plugin setup where turning off internet access breaks dart analyze?

**Comment by incendial** [comment]
@saty9 yes, exactly! I'll check if it has anything related with the DCM. If not, I'll transfer this issue to the Dart team and you repro will be very useful.

**Comment by saty9** [comment]
https://github.com/saty9/dart_analysis_reproduction/commits/master First commit is before I configure analysis to use DCM and `dart analyze` works without internet. 2nd commit sets up DCM and `dart analyze` stops working without internet.

**Comment by incendial** [comment]
Thank you!

**Comment by incendial** [comment]
@saty9 closing this issue since it has been confirmed that the problem comes from the analyzer. See linked issue for more details
